packages/x-components/src/components/global-x-bus.vue

```diff
@@ -0,0 +1,122 @@
+<script lang="ts">
+  import { Component } from 'vue-property-decorator';
+  import { Subscription } from 'rxjs';
+  import { forEach } from '../utils';
+  import { WireMetadata, XEvent, XEventPayload } from '../wiring';
+  import { NoElement } from './no-element';
+
+  type EventListeners = Record<
+    XEvent,
+    (payload: XEventPayload<XEvent>, metadata?: WireMetadata) => unknown
+  >;
```

Comment:
The type is wrong. You can easily see this by invoking any callback with any value. TypeScript should complain.  Also the metadata is not optional.

I suggest to rename it to `XEventListeners`, and make the following change.

```ts
/**
* Map type of every {@link XEvent} and a callback with the payload and metadata for that event.
*/  
type XEventListeners = {
    [Event in XEvent]: (payload: XEventPayload<Event>, metadata: WireMetadata) => void;
  };
```

This change will force you below to cast the payload: `callback(eventPayload as never, metadata);`, we can leave a TODO there remembering us to improve the types. I've been trying to fix this, but I don't know why it infers never. The type is fine this way though, so the issue might be with the `object.ts` utilities. I also tried to improve their types even if it is out of the scope of this PR and failed.